tests/components/dwd_weather_warnings/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+    )
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == DEMO_YAML_CONFIGURATION[CONF_NAME]
+
+    result_data = DEMO_YAML_CONFIGURATION.copy()
+    result_data[CONF_REGION_IDENTIFIER] = result_data.pop(CONF_REGION_NAME)
+
+    assert result["data"] == result_data
+
+
+async def test_import_flow_no_name(hass: HomeAssistant) -> None:
+    """Test a successful import of a YAML configuration with no name set."""
+    data = DEMO_YAML_CONFIGURATION.copy()
+    data.pop(CONF_NAME)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=data
+    )
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == EXPECTED_NAME
+
+    result_data = DEMO_YAML_CONFIGURATION.copy()
+    result_data[CONF_NAME] = EXPECTED_NAME
+    result_data[CONF_REGION_IDENTIFIER] = result_data.pop(CONF_REGION_NAME)
+
+    assert result["data"] == result_data
+
+
+async def test_import_flow_only_required(hass: HomeAssistant) -> None:
+    """Test a successful import of a YAML configuration with only required properties."""
+    data = DEMO_YAML_CONFIGURATION.copy()
+    data.pop(CONF_NAME)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=data
+    )
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == EXPECTED_NAME
+
+    result_data = DEMO_YAML_CONFIGURATION.copy()
+    result_data[CONF_NAME] = EXPECTED_NAME
+    result_data[CONF_REGION_IDENTIFIER] = result_data.pop(CONF_REGION_NAME)
+
+    assert result["data"] == result_data
+
+
+async def test_import_flow_device_already_configured(hass: HomeAssistant) -> None:
+    """Test aborting, if the device is already configured during the import."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=DEMO_CONFIG_ENTRY.copy(),
+        unique_id=DEMO_CONFIG_ENTRY[CONF_REGION_IDENTIFIER],
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=DEMO_YAML_CONFIGURATION.copy()
+    )
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_config_flow_device_already_configured(hass: HomeAssistant) -> None:
+    """Test aborting, if the device is already configured during the config."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=DEMO_CONFIG_ENTRY.copy(),
+        unique_id=DEMO_CONFIG_ENTRY[CONF_REGION_IDENTIFIER],
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.dwd_weather_warnings.config_flow.DwdWeatherWarningsAPI",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=DEMO_CONFIG_ENTRY
+        )
```

Comment:
USER flows do not have any data on init. You should have an init step + a configure step.